File: frontend/src/services/systemMonitor.js
import apiService from './apiService';

export const systemMonitorApi = {
  // 获取系统监控列表
  getSystemMonitors: (params) => {
    return apiService.get('/system/monitor/list', { params })
      .then(res => res.data.data);
  },
  
  // 获取最新的系统监控信息
  getLatestSystemMonitor: () => {
    return apiService.get('/system/monitor/latest')
      .then(res => res.data.data);
  },
  
  // 手动收集系统信息
  collectSystemInfo: () => {
    return apiService.post('/system/monitor/collect')
      .then(res => res.data.data);
  }
}; 

File: backend/api/routes/v1/systemmonitor.go
package v1

import (
	"normaladmin/backend/database"
	"normaladmin/backend/internal/handlers"
	"normaladmin/backend/internal/services"

	"github.com/gin-gonic/gin"
)

// 注册系统监控路由
func RegisterSystemMonitorRoutes(r *gin.RouterGroup) {
	db := database.GetDB()
	systemMonitorService := services.NewSystemMonitorService(db)
	systemMonitorHandler := handlers.NewSystemMonitorHandler(systemMonitorService)

	systemMonitorGroup := r.Group("/system/monitor")
	{
		systemMonitorGroup.POST("/collect", systemMonitorHandler.CollectSystemInfo)
		systemMonitorGroup.GET("/list", systemMonitorHandler.GetSystemMonitors)
		systemMonitorGroup.GET("/latest", systemMonitorHandler.GetLatestSystemMonitor)
	}
}
